src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -1772,6 +1864,82 @@ codet java_string_library_preprocesst::
 
 /*******************************************************************\
 
+Function: java_string_library_preprocesst::make_copy_constructor_code
+
+  Inputs:
+    type - type of the function
+    loc - location in the source
+    symbol_table - symbol table
+
+ Outputs: Code corresponding to:
+           > string_expr = java_string_to_string_expr(arg1)
+           > string_expr_sym = { string_expr.length; string_expr.content }
+           > this = string_expr_to_java_string(string_expr)
+
+ Purpose: Generates code for a constructor of a string object from another
+          string object.
+
+\*******************************************************************/
+
+codet java_string_library_preprocesst::make_copy_constructor_code(
+  const code_typet &type,
+  const source_locationt &loc,
+  symbol_tablet &symbol_table)
+{
+  // Code for the output
+  code_blockt code;
+
+  // String expression that will hold the result
+  string_exprt string_expr=fresh_string_expr(loc, symbol_table, code);
+
+  // Assign argument to a string_expr
+  code_typet::parameterst params=type.parameters();
+  symbol_exprt arg1(params[1].get_identifier(), params[1].type());
+  code.add(code_assign_java_string_to_string_expr(
+    string_expr, arg1, symbol_table));
+
+  // Assign string_expr to symbol to keep track of it
+  exprt string_expr_sym=fresh_string_expr_symbol(loc, symbol_table, code);
+  code.add(code_assignt(string_expr_sym, string_expr));
+
+  // Assign string_expr to `this` object
+  symbol_exprt arg_this(params[0].get_identifier(), params[0].type());
+  code.add(code_assign_string_expr_to_new_java_string(
+    arg_this, string_expr, loc, symbol_table));
+
+  return code;
+}
+
+/*******************************************************************\
+
+Function: java_string_library_preprocesst::make_string_length_code
+
+  Inputs:
+    type - type of the function
+    loc - location in the source
+    symbol_table - symbol table
+
+ Outputs: Code corresponding to:
+           > return this->length
+
+ Purpose: Generates code for the String.length method
+
+\*******************************************************************/
+
+codet java_string_library_preprocesst::make_string_length_code(
+  const code_typet &type,
+  const source_locationt &loc,
+  symbol_tablet &symbol_table)
+{
+  (void) loc; // Avoid compiler warning for unused parameter
```

Comment:
This code has now changed, and the `loc` variable is used. I don't think there are any exceptions that we can handle at this stage.